src/game_state/player.rs

```diff
@@ -72,186 +49,172 @@ pub struct ReportMessage {
-                                z: player_clone.position.z,
-                                yaw: 0,
-                                pitch: 0,
-                                entity_metadata_terminator: 0xff,
-                            })
-                        )
-                        .unwrap();
-                    }
-                })
             }
+        }),
+    }
+}
+
+impl Player {
+    pub fn move_and_look(
+        &mut self,
+        new_position: Option<Position>,
+        new_angle: Option<Angle>,
+    ) -> EntityLookAndMove {
+        if let Some(new_angle) = new_angle {
+            self.angle = new_angle;
+        }
+        let update_packet = self.entity_look_and_move_packet(new_position);
+        if let Some(new_position) = new_position {
+            self.position = new_position;
+        }
+        update_packet
+    }
+
+    pub fn join_game_packet(&self) -> JoinGame {
+        JoinGame {
+            entity_id: self.entity_id,
+            gamemode: 1,
+            dimension: 0,
+            difficulty: 0,
+            max_players: 2,
+            level_type: String::from("default"),
+            reduced_debug_info: false,
+        }
+    }
+
+    pub fn pos_and_look_packet(&self) -> ClientboundPlayerPositionAndLook {
+        ClientboundPlayerPositionAndLook {
+            x: self.position.x,
+            y: self.position.y,
+            z: self.position.z,
+            yaw: 0.0,
+            pitch: 0.0,
+            flags: 0,
+            teleport_id: 0,
+        }
+    }
+
+    fn entity_look_and_move_packet(&self, new_position: Option<Position>) -> EntityLookAndMove {
+        let position_delta = PositionDelta::new(self.position, new_position);
+        EntityLookAndMove {
+            entity_id: self.entity_id,
+            delta_x: position_delta.x,
+            delta_y: position_delta.y,
+            delta_z: position_delta.z,
+            yaw: float_to_angle(self.angle.yaw),
+            pitch: float_to_angle(self.angle.pitch),
+            on_ground: false,
+        }
+    }
+
+    fn player_info_packet(&self) -> PlayerInfo {
+        PlayerInfo {
+            action: 0,
+            number_of_players: 1, //send each player in an individual packet for now
+            uuid: self.uuid.as_u128(),
+            name: self.name.clone(),
+            number_of_properties: 0,
+            gamemode: 1,
+            ping: 100,
+            has_display_name: false,
+        }
+    }
+
+    fn spawn_player_packet(&self) -> SpawnPlayer {
+        SpawnPlayer {
+            entity_id: self.entity_id,
+            uuid: self.uuid.as_u128(),
+            x: self.position.x,
+            y: self.position.y,
+            z: self.position.z,
+            yaw: 0,
+            pitch: 0,
+            entity_metadata_terminator: 0xff,
+        }
+    }
+}
+
+#[derive(Debug, Clone)]
+struct PositionDelta {
+    pub x: i16,
+    pub y: i16,
+    pub z: i16,
+}
+
+impl PositionDelta {
```

Comment:
Now that this is a private module, it might not be necessary, but it's nice that it provides type checking for us. I wouldn't be tremendously against getting rid of it later though